src/resource/file.ts

```diff
@@ -86,12 +88,27 @@ export async function getFile(
  * otherwise.
  * @since 1.1.0
  */
-export async function overwriteFile(
+export async function overwriteFile<T extends File | NodeFile>(
   resourceUrl: UrlString,
-  file: File | Buffer,
+  file: T,
   accessGrant: VerifiableCredential,
   options?: FetchOptions & { contentType?: string }
-) {
+): Promise<T & WithResourceInfo>;
+/**
+ * @deprecated `overwriteFile` should only have `File` input
+ */
+export async function overwriteFile<T extends File | NodeFile | NodeBuffer>(
+  resourceUrl: UrlString,
+  file: T,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions & { contentType?: string }
+): Promise<T & WithResourceInfo>;
+export async function overwriteFile<T extends File | NodeFile | NodeBuffer>(
```

Comment:
```suggestion
export async function overwriteFile<T extends NodeBuffer>(
```
The deprecation notice only needs to apply to the Buffer overload.

Reply:
This set ensures that the deprecation notice applies if I provide something of type `Buffer | File`.

The order of evaluation of the type overloads means that if supply something of *just* `File` or just `NodeFile` then it will be caught by the first overload and the depreciation notice will not apply.

Reply:
After a discussion, I realized my proposed change would actually be a breaking change because no overload would then match the legacy signature (just writing this down for docs puprose).